src/org/daisy/dotify/formatter/impl/row/LeaderManager.java

```diff
@@ -9,24 +9,29 @@
 import java.util.ArrayDeque;
 import java.util.Deque;
 import java.util.logging.Logger;
+import java.util.Optional;
 
 /**
  * TODO: Write java doc.
  */
 class LeaderManager {
     private static final Logger logger = Logger.getLogger(LeaderManager.class.getCanonicalName());
     private Deque<Leader> leaders;
+    private Deque<Optional<String>> leaderModes;
```

Comment:
Could these queues get out of sync, and if so isn't it more appropriate to put the mode as a member of the leader?

Reply:
Normally not, but the mode could be included in the `Leader` object. Or in fact, the `pattern` field could be set to the translated pattern. Yes, I think that's a nicer solution.

Reply:
Addressed this in my latest commit.